src/snowflake/cli/api/commands/snow_typer.py

```diff
@@ -91,12 +92,13 @@ def custom_command(command_callable):
             @wraps(command_callable)
             def command_callable_decorator(*args, **kw):
                 """Wrapper around command callable. This is what happens at "runtime"."""
-                self.pre_execute()
+                execution_id: str = uuid.uuid4().hex
+                self.pre_execute(execution_id)
                 try:
                     result = command_callable(*args, **kw)
                     return self.process_result(result)
                 except Exception as err:
-                    self.exception_handler(err)
+                    self.exception_handler(err, execution_id)
```

Comment:
I'm not familiar with this logic, but I assume that this handler is only invoked for unexpected exceptions? IMHO we should be able differentiate between successful invocations, invocations that failed due to user error, and invocations that failed due to programming errors (the user got a stack trace or something like that).

Reply:
> IMHO we should be able differentiate between successful invocations, invocations that failed due to user error, and invocations that failed due to programming errors

Isn't this possible by checking exception type?

Reply:
to some extent, yes, but IMHO it would be better to avoid relying on a specific set of exception names, since that's not a fixed set. The CLI logic is in the best position to classify exceptions.

Reply:
Hi, I add this [commit](https://github.com/snowflakedb/snowflake-cli/pull/1266/commits/a24161a0c303e5aec3634e9bff05aa8d5d6fba5a) to indicate if an error was handled (expected) or not .